### PR TITLE
Create simple GitHub actions CI

### PR DESCRIPTION
**Pull request by Flova**

### CODE REVIEW

**Comment by Juphex** [comment]
the yml file states "ubuntu-latest".
your readme states Ubunutu 20.04

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: CI
+
+on: [pull_request, workflow_dispatch]
```
Comment (Juphex) [inline .github/workflows/main.yml:3]: I would also suggest to add push on main.
so maybe add:

`push:
         branches: [main]`

**Comment by Flova** [comment]
ROS 1 support officially ends with Ubuntu 20.04. Afterwards, one needs to build everything from source and apply some patches for changed library versions. You can talk to Michael about this as he is trying to maintain ROS 1 past 20.04 (aka. ROS noetic). We are therefore limited to 20.04 :/

As it seems, it is the case that the ROS Industrial pipeline builds on top of docker inside the github actions container. The action container therefore uses the latest Ubuntu, but the docker container that is used to build and test it uses 20.04.

File: .github/workflows/main.yml (hunk 1 above)
Comment (Flova) [reply .github/workflows/main.yml:3]: Due to our settings the main can not be modified without a pull request.

Comment (Juphex) [reply .github/workflows/main.yml:3]: How does pull against master behave?
Is it seen like a push to push when merged?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+jobs:
+  main_ci:
+    runs-on: ubuntu-latest
```
Comment (Juphex) [inline .github/workflows/main.yml:7]: ubuntu-20.04

Comment (Flova) [reply .github/workflows/main.yml:7]: The industrial action runs the docker instance etc. on 22.04 aka latest as described in the documentation. Only the container that is started inside runs 20.04. This is kind of weird I know, but they do it this way. And we have no benefit of running the host system at an older Ubuntu Version.

**Review by Juphex — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (Flova) [reply .github/workflows/main.yml:3]: If I understand you correctly, then yes.

**Pull request merged**